src/glue.rs.in

```diff
@@ -62,13 +87,36 @@ mod _PartitionConsumer {
             inner: Box::pin(run_block_on(consumer.stream(offset.clone()))?)
         })
     }
+    pub fn stream_with_config(
+        consumer: &PartitionConsumer,
+        offset: &Offset,
+        wasm_module_path: &str
+    ) -> Result<PartitionConsumerStream, FluvioError> {
+        let config_wrapper = ConsumerConfigWrapper::new_config_with_wasm_filter(wasm_module_path);
+        let mut builder = ConsumerConfig::builder();
+        builder.smartmodule(Some(SmartModuleInvocation {
+            wasm: SmartModuleInvocationWasm::AdHoc(config_wrapper.wasm_module),
+            kind: SmartModuleKind::Filter,
+            params: Default::default()
+        }));
+        let config = builder.build().expect("Failed to create config");
+        match run_block_on(consumer.stream_with_config(offset.clone(), config)) {
```

Comment:
I feel like a `run_block_on(...).map(|stream| PartitionConsumerStream { inner: Box::pin(stream) })` would be cleaner over a `match` (as the `Err` types match) but don't feel all that strongly about it.

Reply:
Like i said i am not a rust engineer, which way would be considered idiomatic rust?

Reply:
Sorry, just thought my last response comes across as quite blunt.

I would appreciate it if you could maybe point me towards the recommended pattern for this type of code? And if you have time an explanation of why?

I am still learning Rust, enjoying it but still learning 👍

Reply:
Since you are just transforming `Ok,` using `map` is more concise even though `match` works.  See https://doc.rust-lang.org/rust-by-example/error/result/result_map.html.   This is an example of Rust combinator https://learning-rust.github.io/docs/e6.combinators.html

Reply:
Ah that makes sense, thanks for the links, i'll implement that change.